### DISCUSSION TITLE
Unable to mirror RedHat7 packagecloud.io RPMs

### DISCUSSION THREAD

**Discussion by gokuatkai** [rabbitmq/rabbitmq-server#4188] [Other]
Since repository moved to packagecloud.io last year, we are unable to mirror any RPMs from rabbitmq-server repository from packagelcoud:

```
-bash-4.2$ reposync -p /d0/rabbitmq-server -i rabbitmq-server --download-metadata
WARNING: skipping package rabbitmq-server-3.5.7-1.noarch: remote path "../noarch/rabbitmq-server-3.5.7-1.noarch.rpm" not within repodir, unsafe to mirror locally
WARNING: skipping package rabbitmq-server-3.5.8-1.noarch: remote path "../noarch/rabbitmq-server-3.5.8-1.noarch.rpm" not within repodir, unsafe to mirror locally
WARNING: skipping package rabbitmq-server-3.6.0-1.noarch: remote path "../noarch/rabbitmq-server-3.6.0-1.noarch.rpm" not within repodir, unsafe to mirror locally
WARNING: skipping package rabbitmq-server-3.6.1-1.noarch: remote path "../noarch/rabbitmq-server-3.6.1-1.noarch.rpm" not within repodir, unsafe to mirror locally
```

The rabbitmq-erlang repo is perfectly fine tho.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
Our team is not responsible for PackageCloud. It may or may not support mirroring at all.

3.5 and 3.6 packages are so old that my best guess is that something in the repository layout has changed on PackageCloud
in recent years, and the Erlang repository is newer than they are but the RabbitMQ server one is older. We have no control
over this, and there are no repository settings that explicity allow or disallow mirroring.

IIRC this was discussed on rabbitmq-users 6-12 months ago and I could create a local mirror from Cloudsmith. All release artifacts are available for download from GitHub.

**Reply by gokuatkai** [reply]
Maybe you can ask PackageCloud if they might have an idea what is going on? 

ps: 3.5 and 3.6 version are just the first few lines of all rpms package it tries to download

**Comment by lfarkas** [comment]
did you find any solution?

**Reply by gokuatkai** [reply]
The solution is to fix the repo which either Rabbitmq team or PackageCloud is not willing to do.
On the other hand my workaround is just to fix myself the wrong paths - which I ve put in a cron:
```
dir="<path to packages folder>"
repo="rabbitmq-server"
reposync -c /etc/reposync-custom/yum.conf -p ${dir}/tmp -r ${repo} --allow-path-traversal
mv ${dir}/tmp/noarch/*.rpm ${dir}/${repo}/
createrepo ${dir}/${repo}
rm -rf ${dir}/tmp
```

**Comment by lfarkas** [comment]
this PR solve the problem
https://github.com/rpm-software-management/dnf-plugins-core/pull/457
